pkg/yqlib/encoder_kyaml.go

```diff
@@ -0,0 +1,364 @@
+package yqlib
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/color"
+)
+
+type kyamlEncoder struct {
+	prefs KYamlPreferences
+}
+
+func NewKYamlEncoder(prefs KYamlPreferences) Encoder {
+	return &kyamlEncoder{prefs: prefs}
+}
+
+func (ke *kyamlEncoder) CanHandleAliases() bool {
+	// KYAML is a restricted subset; avoid emitting anchors/aliases.
+	return false
+}
+
+func (ke *kyamlEncoder) PrintDocumentSeparator(writer io.Writer) error {
+	if ke.prefs.PrintDocSeparators {
+		if err := writeString(writer, "---\n"); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (ke *kyamlEncoder) PrintLeadingContent(writer io.Writer, content string) error {
```

Comment:
is this the same logic as the yaml encoder? Could it be pulled out into a common function?